worktree.go

```diff
@@ -910,8 +1037,43 @@ func (w *Worktree) Submodules() (Submodules, error) {
 		return nil, err
 	}
 
+	// Pre-fetch origin remote for relative URL resolution to avoid repeated lookups
+	var originURL string
+	for _, s := range m.Submodules {
+		if s.URL != "" && isRelativeURL(s.URL) {
+			originURL, err = w.getOriginRemoteURL()
+			if err != nil {
+				return nil, err
+			}
+			break
+		}
+	}
```

Comment:
We could try to get the default remote, if not found we skip the resolution (by leaving the URL empty):

```suggestion
	var originURL string
	if origin, err := w.r.Remote(DefaultRemoteName); err == nil {
		if origin.c != nil && len(origin.c.URLs) > 0 {
			originURL = origin.c.URLs[0]
		}
	}
```